akka-docs/src/test/java/jdocs/typed/tutorial_5/DeviceManagerProtocol.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.typed.tutorial_5;
+
+import akka.actor.typed.ActorRef;
+
+import java.util.Map;
+import java.util.Set;
+
+abstract class DeviceManagerProtocol {
```

Comment:
Where to define the messages?

Several actors need to handle the same type of messages and I didn't want to introduce complicated conversion between each. All of these go via the `DeviceManager` which delegates to `DeviceGroup` or `DeviceGroupQuery`.

For Java I placed them in a separate `DeviceManagerProtocol`, instead of static classes inside the `DeviceManager`. The main reason is that the messages describe the public API of the actor and it's easier to see if it's separate (and Java is so verbose).

For Scala I placed them in the companion object of the `DeviceManager`.

There are a few other messages that are specific to the `Device` actor, so those are defined in `DeviceProtocol`.

Reply:
:+1: